libraries/ESP8266WebServer/src/ESP8266WebServer.cpp

```diff
@@ -396,46 +388,70 @@ bool ESP8266WebServer::hasArg(String  name) {
 
 
 String ESP8266WebServer::header(String name) {
-  for (int i = 0; i < _headerKeysCount; ++i) {
-    if (_currentHeaders[i].key.equalsIgnoreCase(name))
-      return _currentHeaders[i].value;
+  name.toLowerCase();
```

Comment:
instead of modifying `name`, i would suggest to specialize `map` with a custom comparison operator (which would do `compareIgnoreCase`)